src/components/node-list/styles/_filter.scss

```diff
@@ -63,18 +63,24 @@
   // Fade indicator when unchecked or unset
   &.pipeline-row__toggle-icon--unchecked,
   &.pipeline-row__toggle-icon--unset {
+    // condition 2: semi-bright indicator when checked but not hovered
     opacity: 0.55;
   }
 
-  // Bright indicator when checked or hovered
-  &.pipeline-row__toggle-icon--checked,
-  &.pipeline-row__toggle-icon--unchecked:hover,
-  &.pipeline-row__toggle-icon--unset:hover {
+  &.pipeline-row__toggle-icon--checked {
+    // condition 3: Bright indicator when checked but not hovered
     opacity: 0.75;
+
+    // condition 4: Brightest indicator when checked and hovered on the entire row
+    .pipeline-nodelist__row--visible:hover & {
+      opacity: 1;
+    }
   }
+}
 
-  // Hover states when checked
-  &.pipeline-row__toggle-icon--checked:hover {
+// Hover states for non-filter type icons
+.pipeline-row__toggle-icon--visible:hover {
```

Comment:
Might be better to move this to `_row.scss`

Reply:
good spot - I've moved it across!